src/structures/buffers.rs

```diff
@@ -76,6 +76,10 @@ macro_rules! named_field_buffer_type {
 
             fn try_from(native: $native_type) -> Result<Self> {
                 let mut buffer: $tss_type = Default::default();
+                if native.0.len() > $native_type::MAX_SIZE {
+                    error!("Error: Invalid buffer size(> {})", $native_type::MAX_SIZE);
+                    return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+                }
```

Comment:
To me it was kind of a code smell to have a ```TryFrom``` that could never fail. So this sanity check made the code more consistent which was nice.

Reply:
Actually, that's true, do we *need* it to be `TryFrom`? I don't think it's necessarily a smell, but we could also consider replacing the returned error with [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html)

Reply:
We do not need it to be a ```TryFrom``` because the size of the buffer is validated as the native type is created. Changing the error type to be Infallible would have removed the sense of code smell because that would have shown intent.

Reply:
So is the plan now to change the `Error` type to `Infallible` or keep it as is? The comments seem to imply we agree to use `Infallible` or just `From<T>`, but I don't see a change?
(I'd be fine with either, just want to make sure we're agreed)

Reply:
Sorry I have not been able to work on it that is why you have not seen a change. I thought it looked it nice and consistent to have a size check for all conversions in the macro even though in this conversion it would only have been a sanity check. But I do not really care that much. When I find the time I will look if it can be a ```From<T>``` conversion otherwise I will change the error to ```Infallible```.

Reply:
It has now been changed to a ```From``` conversion.